yarn-project/circuit-types/src/tx/simulated_tx.ts

```diff
@@ -45,6 +48,29 @@ function processOutputFromJSON(json: any): ProcessOutput {
 export class SimulatedTx {
   constructor(public tx: Tx, public privateReturnValues?: ProcessReturnValues, public publicOutput?: ProcessOutput) {}
 
+  /**
+   * Returns suggested total and teardown gas limits for the simulated tx.
+   * Note that public gas usage is only accounted for if the publicOutput is present.
+   * @param pad - Percentage to pad the suggested gas limits by, defaults to 10%.
+   */
+  public getGasLimits(pad = 0.1) {
```

Comment:
What do you think about instead having a `getGasUsed` that returns the true gas used, and have a separate function that scales/pads?

Reply:
Sure! I was going to consume this from aztecjs, but it's true that this padding shouldn't be a concern of `SimulatedTx`.